Extensions/Video/videoruntimeobject.js

```diff
@@ -0,0 +1,241 @@
+  this._renderer.updateHeight();
+};
+
+gdjs.VideoRuntimeObject.prototype.getWidth = function() {
+  return this._width;
+};
+
+gdjs.VideoRuntimeObject.prototype.getHeight = function() {
+  return this._height;
+};
+
+gdjs.VideoRuntimeObject.prototype.play = function() {
+  this._renderer.play();
+};
+
+gdjs.VideoRuntimeObject.prototype.pause = function() {
+  this._renderer.pause();
+};
+
+gdjs.VideoRuntimeObject.prototype.setLoop = function(bool) {
+  this._renderer.setLoop(bool);
+};
+
+gdjs.VideoRuntimeObject.prototype.mute = function(bool) {
+  this._renderer.setMute(bool);
+};
+
+gdjs.VideoRuntimeObject.prototype.isMuted = function() {
+  return this._renderer.isMuted();
+};
+
+/* Tool normalize for value 
+
+input 0-1, return a number 0-100
+this._normalize(volume, 1, 0)*100;
+
+input 0-100, return output 0-1
+this._normalize(parseFloat(number), 100, 0)
+*/
+
+gdjs.VideoRuntimeObject.prototype._normalize = function(val, min, max) {
+  return (val - min) / (max - min);
+};
+
+/**
+ * Limit value in an interval
+ */
+
+gdjs.VideoRuntimeObject.prototype._clamp = function(val, min, max) {
+  return val <= min ? min : val >= max ? max : val;
+};
+
+gdjs.VideoRuntimeObject.prototype.setVolume = function(newVolume) {
+  //newVolume = 0-100 , newVolume = normalize(newVolume) = 0-1, clamp = clamp(newVolume) = 0-1
+  var _newVolume = this._clamp(this._normalize(newVolume, 0, 100), 0, 1);
+  this._renderer.setVolume(_newVolume);
+};
+
+gdjs.VideoRuntimeObject.prototype.getVolume = function() {
+  return this._normalize(this._renderer.getVolume(), 0, 1) * 100;
+};
+
+gdjs.VideoRuntimeObject.prototype.isPlayed = function() {
+  return this._renderer.isPlayed();
+};
+
+gdjs.VideoRuntimeObject.prototype.isPaused = function() {
+  return !this._renderer.isPlayed();
+};
+
+gdjs.VideoRuntimeObject.prototype.isLooped = function() {
+  return this._renderer.isLooped();
+};
+
+gdjs.VideoRuntimeObject.prototype.controlsAreShowing = function() {
+  return this._renderer.controlsAreShowing();
+};
+
+gdjs.VideoRuntimeObject.prototype.getDuration = function() {
+  return this._renderer.getDuration();
+};
+
+gdjs.VideoRuntimeObject.prototype.isEnded = function() {
+  return !this._renderer.isEnded();
+};
+
+gdjs.VideoRuntimeObject.prototype.setCurrentTime = function(number) {
+  this._renderer.setCurrentTime(number);
+};
+
+gdjs.VideoRuntimeObject.prototype.getCurrentTime = function() {
+  return this._renderer.getCurrentTime();
+};
+
+gdjs.VideoRuntimeObject.prototype.setPlaybackSpeed = function(playbackSpeed) {
+  var new_playbackSpeed = this._clamp(
+    this._normalize(playbackSpeed, 0, 100),
+    0,
+    1
+  );
+  this._renderer.setPlaybackSpeed(new_playbackSpeed); // Tell the renderer to update the rendered object
+};
+
+/**
+ * Get playblack speed.
+ * return 0-100
+ */
+gdjs.VideoRuntimeObject.prototype.getPlaybackSpeed = function() {
+  return this._normalize(this._renderer.getPlaybackSpeed(), 0, 1) * 100;
+};
+
+//NOTE Controls impossible to display ?
```

Comment:
I don't think they will be displayed, because we're displaying a texture, which can't have controls. Let's forget about controls. Anyway, nobody will want to display them, or if they want they will create their own buttons